Toxygates/src/t/server/viewer/intermine/Intermine.scala

```diff
@@ -67,71 +67,100 @@ class IntermineConnector(instance: IntermineInstance,
-   */
-  def addProbeLists(ls: ListService, lists: Iterable[StringList], replace: Boolean): Unit =
-    for { l <- lists } {
-      addProbeList(ls, l.items(), l.name(), replace)
-    }
-
   /**
    * Add a probe list to InterMine by first mapping it into genes (lazily)
    */
-  def addProbeList(ls: ListService, probes: Iterable[String], name: String, replace: Boolean,
+  def addProbeList(ls: ListService, list: GeneList, replace: Boolean,
     tags: Seq[String] = Seq("panomicon")): Option[ItemList] =
     addEntrezList(ls,
-          platforms.resolve(probes.toSeq).flatMap(_.genes.map(_.identifier)),
-          name, replace, tags)
+      list.copy(items =
+        platforms.resolve(list.items).flatMap(_.genes.map(_.identifier))
+      ),
+      replace, tags)
 
   /**
    *  Add a list of NCBI/Entrez genes to InterMine.
    *  @param ls the ListService
-   *  @param genes the genes (entrez IDs) to be added
-   *  @param name the name of the list to be created
+   *  @param list the genes (entrez IDs) to be added
    *  @param replace whether any existing list with the same name should be replaced
    *  @param tags any tags to be added to the list (stored on the InterMine server)
    *  @return the ItemList if it was successfully created, or None if the operation failed.
    */
-  def addEntrezList(ls: ListService, genes: Seq[String], name: String, replace: Boolean,
-    tags: Seq[String] = Seq("panomicon")): Option[ItemList] = {
+  def addEntrezList(ls: ListService, list: GeneList, replace: Boolean,
+                    tags: Seq[String] = Seq("panomicon")): Option[ItemList] = {
 
-    for {existingList <- Option(ls.getList(name))} {
+    for {existingList <- Option(ls.getList(list.name))} {
       if (replace) {
         println(s"Delete list $existingList for replacement")
         ls.deleteList(existingList)
       } else {
         //The list existed and replacement was not requested, so we can't proceed.
         //Could report this message to the user somehow
-        println(s"Not exporting list '$name' since it already existed (replacement not requested)")
+        println(s"Not exporting list '${list.name}' since it already existed (replacement not requested)")
         //throw new IntermineException(
         //  s"Unable to add list, ${name.get} already existed and replacement not requested
         return None
       }
     }
 
-    val ci = new ls.ListCreationInfo("Gene", name)
+    val ci = new ls.ListCreationInfo("Gene", list.name)
     //Note: we have the option of doing a fuzzy (e.g. symbol-based) export here
-    ci.setContent(genes.asJava)
+    ci.setContent(list.items.asJava)
     ci.addTags(tags.asJava)
-    println(s"Exporting list '$name'")
+    println(s"Exporting list '${list.name}'")
     Some(ls.createList(ci))
   }
+
+  def exportLists(user: String, pass: String, lists: Iterable[GeneList], replace: Boolean): Unit =
+    try {
+      val ls = getListService(Some(user), Some(pass))
```

Comment:
I have only one issue with providing the user's password... AFAIK to avoid this, intermine provides the use of access tokens, that can be used as an alternative to the password, as they are time-limited (although also potentially permanent) ways to access the lists.

For example, with the python API for Intermine, rather than initializing a service using user/pass as shown below:
`service = Service(service, username=user, password=pwd)`
you would do it with
`service = Service(service, token=token)`

After defining the service, you actually access/create list using appropriate methods on the defined service.. something along the lines of 
`service.create_list(genes, name=lname, list_type='Gene')`
(which is what you are doing here in line 131... the initialization of the service I didnt see it/find it :sweat_smile:  

So I was wondering if (it wasnt too much work) it would be possible to change the access method

PS: first time using this comments.. so not sure how its going to show up in the review of the PR XD

Reply:
Hi Rodolfo, thanks for your comment. I have thought a little about this before and internally, we do use tokens for some of the communication once the session has been established. 
If we use tokens throughout and never send login details, as you suggest, the end user would be responsible for creating and supplying a token, right? Do you think that would be easy for average users to do? My impression is that this would be a more complicated experience than supplying username/password, but I might be wrong.